### ISSUE TITLE
Dropping a keyspace with nested UDT results in an Error and keyspace become "zombie"

### ISSUE THREAD

**Issue by roydahan** [scylladb/scylla#2490]
*Installation details*
Scylla version (or git commit hash): 1.7

After creating the below tables with some UDTs, trying to drop the keyspace results in error:
**InvalidRequest: Error from server: code=2200 [Invalid query] message="Cannot drop user type udttests.test_can_insert_nested_collections as it is still used by user type test_can_insert_nested_collections_nested**

Describe schema still shows the keyspace and it's still functioning.
When trying to drop it again, the command supposedly succeed, the keyspace doesn't appear anymore in "DESC SCHEMA" but you can still use it (USE udttests) and trying to recreate it shows it as "Already exist".
However, you cannot select from it (although you can create a table in it)...
cqlsh:udttests> use udttests;
cqlsh:udttests> create table roy (k int PRIMARY KEY);
AlreadyExists: Table 'udttests.roy' already exists
cqlsh:udttests> create table roy1 (k int PRIMARY KEY);
cqlsh:udttests> select * from udttests.roy;
Keyspace 'udttests' not found.
cqlsh:udttests> select * from udttests.roy1;
Keyspace 'udttests' not found.

**To reproduce:**

```
[cqlsh 5.0.1 | Cassandra 2.2.8 | CQL spec 3.3.1 | Native protocol v4]
Use HELP for help.
cqlsh> desc SCHEMA 

cqlsh> CREATE KEYSPACE udttests WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'};
cqlsh> 
cqlsh> USE udttests;
cqlsh:udttests> 
cqlsh:udttests>             CREATE TYPE test_can_insert_nested_collections (
            ...                 m frozen<map<int,text>>,
            ...                 t tuple<int,text>,
            ...                 l frozen<list<int>>,
            ...                 s frozen<set<int>>
            ...             );
cqlsh:udttests> 
cqlsh:udttests>             CREATE TYPE test_can_insert_nested_collections_nested (
            ...                 m frozen<map<int,text>>,
            ...                 t tuple<int,text>,
            ...                 l frozen<list<int>>,
            ...                 s frozen<set<int>>,
            ...                 u frozen<test_can_insert_nested_collections>
            ...             );
cqlsh:udttests> 
cqlsh:udttests>             CREATE TABLE test_can_insert_nested_collections (
            ...                 k int PRIMARY KEY,
            ...                 map_map map<frozen<map<int,int>>, frozen<map<int,int>>>,
            ...                 map_set map<frozen<set<int>>, frozen<set<int>>>,
            ...                 map_list map<frozen<list<int>>, frozen<list<int>>>,
            ...                 map_tuple map<frozen<tuple<int, int>>, frozen<tuple<int>>>,
            ...                 map_udt map<frozen<test_can_insert_nested_collections_nested>, frozen<test_can_insert_nested_collections>>,
            ...             );
cqlsh:udttests> 
cqlsh:udttests> DROP KEYSPACE udttests;
InvalidRequest: Error from server: code=2200 [Invalid query] message="Cannot drop user type udttests.test_can_insert_nested_collections as it is still used by user type test_can_insert_nested_collections_nested"
cqlsh:udttests> DROP KEYSPACE udttests;
cqlsh:udttests> CREATE KEYSPACE udttests WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'};
AlreadyExists: Keyspace 'udttests' already exists
cqlsh:udttests> 
```

When creating the table without the nested type all works smoothly.

**Comment by duarten** [comment]
This is solved in master by scylladb/scylla@6c8b5fc09d. I can backport that part if needed.

**Comment by slivne** [comment]
Larisa / Roy which version are you running - can you check if you have the
above patch in your code.

I am not sure its crtical to backport this to 1.7 ... we should have it
fixed in master for the next release

**Comment by duarten** [comment]
The Scylla version is set to 1.7 in the description, and looking at the code, it can only happen in versions below master. I sent a fix to 1.7 :)

**Comment by larisau** [comment]
I don't have the above patch. Anyway I'm going to build scylla, so I'll run the test and update the issue.

**Comment by duarten** [comment]
Fixed by 	60af7ea.
